### PR TITLE
Feature: AverageTrueRange

### PR DESCRIPTION
**Pull request by sgoley**
Motivation of PR: 
https://github.com/enzoampil/fastquant/issues/275

Description of intended feature:
https://www.investopedia.com/terms/a/atr.asp

New Strategy added:
ATRStrategy

Strategy Arguments: 
- Period (Int)
- Multiplier (Float)

### CODE REVIEW

**Comment by jbdelmundo** [comment]
Hi @sgoley 
You also need to import your strategy in `fastquant/strategies/__init__.py` to fix the import error on tests

File: python/fastquant/strategies/atr.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        print("===Strategy level arguments===")
+        print("atr_period :", self.atr_period)
+        print("atr_factor :", self.atr_factor)
+
+        self.atr = bt.indicators.AverageTrueRange(period=self.atr_period)
+        self.atr_factored = bt.indicators.AverageTrueRange(period=self.atr_period) * self.atr_factor
+
+        self.dataclose = self.datas[0].close
+        self.dataopen = self.datas[0].open
+        self.lastdataclose = self.datas[1].close
```
Comment (enzoampil) [inline python/fastquant/strategies/atr.py:48]: @sgoley I think you may have meant `self.datas[-1]` if you were intending to utilize the previous close?

Comment (enzoampil) [reply python/fastquant/strategies/atr.py:48]:
```suggestion
        self.lastdataclose = self.datas[-1].close
```

File: python/fastquant/strategies/atr.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.lastdataopen = self.datas[1].open
```
Comment (enzoampil) [inline python/fastquant/strategies/atr.py:49]:
```suggestion
        self.lastdataopen = self.datas[-1].open
```

Comment (enzoampil) [reply python/fastquant/strategies/atr.py:49]: Same comments as dataclose

**Review by enzoampil — CHANGES_REQUESTED** [review]
Thanks for this @sgoley , and for providing an issue as context, and an investopedia article to elaborate to strategy :smile:

I'm generally okay, but let's start with fixing the build error due to the wrong indexing. I'll see if it's quick to resolve. For now, kindly note the lines I'm referring to in the comments :smile:

Also, you'll see that I made some corrections to the indexing, mainly around getting the "last" data point using `-1` instead of `1`. Also, you'll see that created a new `set_trailing_stop` method that ensures that the indexing only runs during the actual strategy run (called during `buy_signal` and `sell_signal`.

**Comment by sgoley** [comment]
I'm trying to see how I can build this and test via a local env before merging. 
Is the best way to do that something like the standard `python3 setup.py sdist bdist_wheel` and then install through pip locally from the wheel?
Thanks.

**Comment by enzoampil** [comment]
@sgoley Yes, that's one way to do it.

Might be faster to just navigate to run your script / jupyter notebook from `fastquant/python`.

You should be able to import normally from here.

Guideline is found in [CONTRIBUTING.md](https://github.com/enzoampil/fastquant/blob/master/CONTRIBUTING.md#get-started)

**Comment by enzoampil** [comment]
@sgoley Any update with this? Happy to help out however I can to get this merged :smile:

**Pull request closed without merging**